modules/tinymce/src/plugins/table/test/ts/browser/command/ModifyClassesCommandsTest.ts

```diff
@@ -60,22 +65,92 @@ describe('browser.tinymce.plugins.table.command.ModifyClassesCommandsTest', () =
-        editor.setContent(contentWithoutClass);
-        TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1);
+      context('With a selection of one', () => {
+        it('TINY-7163: Can be toggled on', () => {
+          const editor = hook.editor();
+          setContentAndSelection(editor, contentWithoutClass, 0, 1);
 
-        execCmdAndAssertEvent(editor, 'mceTableCellToggleClass', 'a');
-        TinyAssertions.assertContent(editor, contentWithClass);
+          execCmdAndAssertEvent(editor, 'mceTableCellToggleClass', 'a');
+          TinyAssertions.assertContent(editor, contentWithClass);
+        });
+
+        it('TINY-7163: Can be toggled off', () => {
+          const editor = hook.editor();
+          setContentAndSelection(editor, contentWithClass, 0, 1);
+
+          execCmdAndAssertEvent(editor, 'mceTableCellToggleClass', 'a');
+          TinyAssertions.assertContent(editor, contentWithoutClass);
+        });
       });
 
-      it('TINY-7163: Can be toggled off', () => {
-        const editor = hook.editor();
-        editor.setContent(contentWithClass);
-        TinySelections.setSelection(editor, [ 0, 0, 0 ], 0, [ 0, 0, 0 ], 1);
+      context('With a selection of several cells', () => {
+        const contentWithDoubleClass = (
+          '<table>' +
+            '<tbody>' +
+              '<tr>' +
+                '<td class="a">1</td>' +
+                '<td class="a">2</td>' +
+              '</tr>' +
+            '</tbody>' +
+          '</table>'
+        );
 
-        execCmdAndAssertEvent(editor, 'mceTableCellToggleClass', 'a');
-        TinyAssertions.assertContent(editor, contentWithoutClass);
+        const contentWithoutClassDoubleSelection = (
+          '<table>' +
+            '<tbody>' +
+              '<tr>' +
+                '<td data-mce-first-selected="1" data-mce-selected="1">1</td>' +
+                '<td data-mce-last-selected="1" data-mce-selected="1">2</td>' +
+              '</tr>' +
+            '</tbody>' +
+          '</table>'
+        );
+
+        const contentWithMixedClassDoubleSelection = (
+          '<table>' +
+            '<tbody>' +
+              '<tr>' +
+                '<td data-mce-first-selected="1" data-mce-selected="1" class="a">1</td>' +
+                '<td data-mce-last-selected="1" data-mce-selected="1">2</td>' +
+              '</tr>' +
+            '</tbody>' +
+          '</table>'
+        );
+
+        const contentWithClassDoubleSelection = (
+          '<table>' +
+            '<tbody>' +
+              '<tr>' +
+                '<td data-mce-first-selected="1" data-mce-selected="1" class="a">1</td>' +
+                '<td data-mce-last-selected="1" data-mce-selected="1" class="a">2</td>' +
+              '</tr>' +
+            '</tbody>' +
+          '</table>'
+        );
+
+        it('TINY-7691: Can be toggled on', () => {
+          const editor = hook.editor();
+          editor.setContent(contentWithoutClassDoubleSelection);
```

Comment:
Can we please use your `setContentAndSelection` function here, as this should be setting a selection otherwise it's not a realistic test. Remember that normally we have the attributes and the selection sitting inside one of the cells.

- [ ] Ensure the native selection is set as well for these tests